### PR TITLE
Button Component

### PR DESCRIPTION
**Pull request by serling**

### CODE REVIEW

File: components/button/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## Autor:
+
+Thomas Christoffersen
+
+## Dependencies:
+
+- classnames
+- prop-types
+- @creuna/lib Icon
```
Comment (asbjornh) [inline components/button/README.md]: Same comment as [here](https://github.com/Creuna-Oslo/react-components/pull/7/files/ca85d18acb169606aa785fafc645153eb88df03e#diff-e83f172eff764b1e5b97d8a73588b330). This shouldn't reference something that looks like an NPM package that doesn't exist

File: components/button/button.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cn from "classnames";
+
+import Icon from "../icon";
+
+const themes = {
```
Comment (asbjornh) [inline components/button/button.jsx]: I think themes are too specific. People can add themes if they need them and I think they should be removed :)

File: components/button/button.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        className="button__icon"
+        name={isActive ? iconNameActive : iconName}
+        size={iconSize}
+      />
+    )}
+  </button>
+);
+
+Button.propTypes = {
+  children: PropTypes.oneOfType([
```
Comment (asbjornh) [inline components/button/button.jsx]: `PropTypes.node` should be enough here (but do check)

File: components/button/README.md (hunk 1 above)
Comment (serling) [reply components/button/README.md]: done

File: components/button/button.jsx (hunk 3 above)
Comment (serling) [reply components/button/button.jsx]: Looks like PropTypes.node is all it takes! Fixed,

File: components/button/button.jsx (hunk 2 above)
Comment (asbjornh) [reply components/button/button.jsx]: I absolutely agree about themes being a great pattern! My issue is not with the fact that the component has themes but with the specific included themes. I find it highly likely that the theme called `big` with a padding of `2rem` will have to be deleted every time you download the component. And the `default` theme has no styling at all.

My concern is reducing the chore of having to delete the same stuff every time. In some components it makes sense to ship with themes, like the Link component with the 'underline' theme which is very general and probably useful.

Finally, I think there are better ways of spreading the good word of nice practices and patterns than pushing them where they're not needed, which is a bad practice IMO

Comment (serling) [reply components/button/button.jsx]: Haha, I agree - horrible theme examples on my part. I'll add some sensible styling to it! In general though, I have no problems with pushing some cool ideas and nice practices into our components. At least until this knowledge base of yours is in place and we actually have a forum to "spread the good word".

Comment (asbjornh) [reply components/button/button.jsx]: @Reasonable-Solutions Do you have an opinion on this?

Comment (Reasonable-Solutions) [reply components/button/button.jsx]: > My concern is reducing the chore of having to delete the same stuff every time. In some components it makes sense to ship with themes, like the Link component with the 'underline' theme which is very general and probably useful.

Sensible defaults, 👍 

I really have to keep in mind when looking at this code that it's a bunch of files in any of our projects and not an npm module.

Comment (Reasonable-Solutions) [reply components/button/button.jsx]: it being files makes the api much less sensitive and iterating on "what works well" is real fast too.

**Pull request merged**